# Identifying Differentially Expressed Genes, Methylation Sites and Interfering RNAs in Human Genome in relation to pathologies : Application on Multiple Sclerosis Patients 

# The Project aim:
Creat an R  script that perform a full genomic, transcriptomic and methylation study  and filter the target gene specific data .

Several researche  have proved that the genomic, transcriptomic and the methylation profils are affected with environmental factors, which creat different prevalence area around the world,

# Different prevalence:

To solve the problem of different prevalence that can be correlated with different genomic, transcriptomic and epigenomic profiles; 
In this study, we will use data from each prevalence area.


# The bioinformatic workflow
![workflow](https://user-images.githubusercontent.com/73958439/163676806-83611165-1435-47a1-822c-82b354d1ad5b.jpg)

# Case study





# Suplementary data





# Workflow reproducibility



# Result 

The genetic specific data will be automatically exported in the  working directory folder as 4 txt files:

- The gene expression profile : p value and LogFC

- If the gene is : overexpressed , underexpressed or not differentially expressed.

- List of the  CpG sites that are methylated in the gene specific promoter with supplementary data mainly the B value , DMR (differential methylation or not ), and  the relation to Cpg island

- List of miRNAs that target the gene in healthy condition

- Are these miRNA expressed in disease group or not ( the user need to search for each miRNA separately ).


# Contact for data update 
Tech leader : Hiba Ben Aribi
email : benaribi.hiba@gmail.com


## Team 
Hiba Ben Aribi, Master in Neuroscience and Biotechnology , UTM

Guerbouj Souheila,Phd in Genetics and molecular biology ,IPT,UTM

Farah Ayadi, Master in molecular biology ,UTM

Careen Naitore,Master in Bioinformatics and molecular biology, JKUAT



